staging/kos/pkg/controllers/connectionagent/connectionagent.go

```diff
@@ -127,6 +125,39 @@ type attachmentState struct {
 	ifc        netfabric.NetworkInterface
 }
 
+// attVNIsIntel represents the "intel" on the attachment current location (in
+// terms of virtual network). It stores the VNIs with whom an attachment has been
+// seen by the notification handlers on remote attachments.
+// It's NOT THREAD-SAFE.
+type attVNIsIntel struct {
+	onlyVNI uint32
+	vnis    map[uint32]struct{}
+}
+
+func newAttVNIsIntel() *attVNIsIntel {
+	return &attVNIsIntel{
+		vnis: make(map[uint32]struct{}, 1),
+	}
+}
+
+func (avi *attVNIsIntel) addVNI(vni uint32) {
+	avi.vnis[vni] = struct{}{}
+	if len(avi.vnis) == 1 {
+		avi.onlyVNI = vni
+	}
+}
+
+func (avi *attVNIsIntel) removeVNI(vni uint32) {
+	delete(avi.vnis, vni)
+	if len(avi.vnis) == 1 {
+		avi.onlyVNI = vni
```

Comment:
`vni` is not the one remaining member of the set.

Reply:
Apologies for the silly mistake, fixing it now.

Reply:
Fixed in 02cd91d.